test/integration/main_test.go

```diff
@@ -616,13 +547,17 @@ func testUploadFile(ctx context.Context, tb testing.TB, bucket, object string, d
 		server.MetadataKeyWorkflowRunID:              testWorkflowRunID,
 	}
 
-	if _, err := io.Copy(wc, data); err != nil {
-		return fmt.Errorf("failed to copy bytes: %w", err)
-	}
-	if err := wc.Close(); err != nil {
-		return fmt.Errorf("failed to close writer: %w", err)
+	closer, err := testutil.UploadGCSFile(ctx, gcsClient, bucket, object, data, metadata)
+	if err != nil {
+		return fmt.Errorf("failed to upload file: %w", err)
 	}
 
+	tb.Cleanup(func() {
+		if err := closer(); err != nil {
```

Comment:
What will happen if closer is nil here?

Reply:
Fixed.

Reply:
Just blindly return closer as non-nil doesn't fix the issue. For example, if you fail to create gcs object and then you return closer as non-nil, when you call closer(), it will throw error which brings noise.